api/src/main/java/org/cardanofoundation/rosetta/api/network/service/TopologyConfigServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package org.cardanofoundation.rosetta.api.network.service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.openapitools.client.model.Peer;
+
+import org.cardanofoundation.rosetta.api.network.model.Producer;
+import org.cardanofoundation.rosetta.api.network.model.PublicRoot;
+import org.cardanofoundation.rosetta.api.network.model.TopologyConfig;
+import org.cardanofoundation.rosetta.common.exception.ExceptionFactory;
+import org.cardanofoundation.rosetta.common.util.FileUtils;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class TopologyConfigServiceImpl implements TopologyConfigService {
+
+  @Value("${cardano.rosetta.TOPOLOGY_FILEPATH}")
+  private String topologyFilepath;
+
+  private final Object lock = new Object();
+  private List<Peer> cachedPeers;
+
+  @PostConstruct
+  public void filePathExistingValidator() {
+    FileUtils.validator(topologyFilepath);
+  }
+
+  @Override
+  public List<Peer> getPeers() {
+    List<Peer> peers = cachedPeers;
+    if (peers == null) {
+      synchronized (lock) {
+        peers = cachedPeers;
+        if (peers == null) {
+          TopologyConfig topologyConfig = loadTopologyConfig();
```

Comment:
If we get peers only from the file, it's better to make it initialized with @PostConstruct. Or don't we have the file yet after bean initialization?

Reply:
Done